domains/runtime/evm/src/lib.rs

```diff
@@ -140,7 +141,20 @@ impl fp_self_contained::SelfContainedCall for RuntimeCall {
         len: usize,
     ) -> Option<TransactionValidity> {
         match self {
-            RuntimeCall::Ethereum(call) => call.validate_self_contained(info, dispatch_info, len),
+            RuntimeCall::Ethereum(call) => {
+                // Ensure the caller can pay for the consensus chain storage fee
+                let storage_fee = DomainTransactionByteFee::get()
```

Comment:
I'm confused why this change is required here. As long as on charge transaction of the transaction_payment pallet is adjusted to collect the storage fee along side the fees for domain compute and transaction_byte fees, this change wont be required and should work out of the box

Reply:
The evm transaction have their own way to validate and `pre_dispatch` tx and they won't touch the `pallet-transaction-payment` like the substrate extrinsic does, I didn't find any other better interfaces export from frontier to charge storage fee so have to use this one. Plz see:
https://github.com/subspace/frontier/blob/2b09a676cac8cca665c882a28bd13b2acef39395/primitives/self-contained/src/checked_extrinsic.rs#L73-L164